### PR TITLE
[readme-gen] Generate README for main endpoints

### PR DESCRIPTION
**Pull request by jaylees14**
Here's my first stab at generating a README.
Couple of things that need adding:
- Structs
- A description of the file structure
- A general introduction to the generated files?

When reviewing this PR, please look at a rich diff (e.g.: https://github.com/TempleEight/temple/blob/5ad4cee9ac1fe6a8d7168450ccdfc027f4971cde/src/e2e/resources/simple-temple-expected/README.md) and see what you think of sizing etc.

We can change anything, and hopefully my code is fairly modular so should be easy to move things around 😉

### CODE REVIEW

File: src/main/scala/temple/generate/docs/DocumentationGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  // Add general information about the endpoint, including URL, method, description & auth if used
+  private def addGeneralInfo(
+    builder: MarkupDocBuilder,
+    url: String,
+    method: String,
+    description: String,
+    usesAuth: Boolean,
+  ): Unit = {
+    // Have to add whitespace manually :'(
```
Comment (c-harding) [inline src/main/scala/temple/generate/docs/DocumentationGenerator.scala:24]: 😢

**Review by Lewiky — DISMISSED** [review]
Looks good, personally happy with the way it looks!

File: src/main/scala/temple/generate/docs/internal/DocumentationGeneratorUtils.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  private[docs] def httpCodeString(code: Int): String = code match {
+    case 200 => "200 OK"
+    case 302 => "302 FOUND"
+    case 400 => "400 BAD REQUEST"
+    case 401 => "401 UNAUTHORIZED"
+    case 403 => "403 FORBIDDEN"
+    case 404 => "404 NOT FOUND"
+    case 500 => "500 INTERNAL SERVER ERROR"
+    case v   => throw new RuntimeException(s"Unhandled HTTP Code $v")
```
Comment (c-harding) [inline src/main/scala/temple/generate/docs/internal/DocumentationGeneratorUtils.scala]: v?

**Review by c-harding — DISMISSED** [review]
this looks stunning 😍

File: src/main/scala/temple/generate/docs/internal/DocumentationGeneratorUtils.scala (hunk 2 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/docs/internal/DocumentationGeneratorUtils.scala]: v for the first key that jay happened to press on his keyboard 😉

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

nice, imagine writing these by hand

**Review by c-harding — APPROVED** [review]

**Pull request merged**